packages/mds-audit/attachments.ts

```diff
@@ -0,0 +1,177 @@
+const s3Bucket = env.ATTACHMENTS_BUCKET
+const s3BucketSubdir = env.ATTACHMENTS_SUBDIR
+const s3Region = env.ATTACHMENTS_REGION
+const s3ACL = env.ATTACHMENTS_ACL
+const s3 = new aws.S3()
+const memoryStorage = multer.memoryStorage()
+
+/* eslint-disable-next-line */
+aws.config.getCredentials(async (err) => {
+  if (err) {
+    await log.error('Error getting AWS credentials', err.stack || err)
+  } else if (aws.config.credentials) {
+    aws.config.update({
+      secretAccessKey: aws.config.credentials.secretAccessKey,
+      accessKeyId: aws.config.credentials.accessKeyId,
+      region: s3Region
+    })
+  }
+})
+
+export const multipartFormUpload = multer({ storage: memoryStorage }).single('file')
+
+export function attachmentSummary(attachment: Attachment): AttachmentSummary {
+  const thumbnailUrl = attachment.thumbnail_filename ? attachment.base_url + attachment.thumbnail_filename : ''
+  return {
+    attachment_id: attachment.attachment_id,
+    attachment_url: attachment.base_url + attachment.attachment_filename,
+    thumbnail_url: thumbnailUrl
+  }
+}
+
+export function validateFile(file: Express.Multer.File) {
+  if (!file || file.buffer.byteLength === 0) {
+    throw new ValidationError('No attachment found')
+  } else if (path.extname(file.originalname).replace('.', '') === '') {
+    throw new ValidationError(`Missing file extension in filename ${file.originalname}`)
+  } else if (!supportedMimetypes.includes(file.mimetype)) {
+    throw new UnsupportedTypeError(`Unsupported mime type ${file.mimetype}`)
+  }
+}
+
+async function writeAttachmentS3(file: Express.Multer.File) {
+  // Process attachment and thumbnail and output to Buffers
+  const [attachmentBuf, thumbnailBuf] = await Promise.all([
+    sharp(file.buffer)
+      .rotate() // automatically based on EXIF
+      .toBuffer(),
+    sharp(file.buffer)
+      .rotate()
+      .resize(thumbnailSize, thumbnailSize, {
+        fit: sharp.fit.inside,
+        withoutEnlargement: true
+      })
+      .toBuffer()
+  ])
+
+  // Upload attachment and thumbnail to S3
+  const attachmentId = uuid()
+  const ext = path.extname(file.originalname).toLowerCase()
+  const attachmentFilename = attachmentId + ext
+  const thumbnailFilename = `${attachmentId}.thumbnail${ext}`
+  await Promise.all([
+    s3
+      .upload({
+        Bucket: s3Bucket,
+        Key: [s3BucketSubdir, attachmentFilename].join('/'),
+        Body: attachmentBuf,
+        ACL: s3ACL
+      })
+      .promise(),
+    s3
+      .upload({
+        Bucket: s3Bucket,
+        Key: [s3BucketSubdir, thumbnailFilename].join('/'),
+        Body: thumbnailBuf,
+        ACL: s3ACL
+      })
+      .promise()
+  ])
+  return {
+    attachment_filename: attachmentFilename,
+    attachment_id: attachmentId,
+    base_url: `https://${s3Bucket}.s3-${s3Region}.amazonaws.com/${s3BucketSubdir}/`,
+    mimetype: file.mimetype,
+    thumbnail_filename: thumbnailFilename,
+    thumbnail_mimetype: file.mimetype
+  } as Attachment
```

Comment:
Why the need to cast here?

Reply:
It was because `recorded` wasn't included in the return. Fixed by making `recorded` optional.